src/Data/AppendMap.hs

```diff
@@ -0,0 +1,417 @@
+isProperSubmapOf = coerce (Map.isProperSubmapOf @k @a)
+
+isProperSubmapOfBy :: Ord k => (a -> b -> Bool) -> AppendMap k a -> AppendMap k b -> Bool
+isProperSubmapOfBy f = coerce (Map.isProperSubmapOfBy f)
+
+lookupIndex :: forall k a. Ord k => k -> AppendMap k a -> Maybe Int
+lookupIndex = coerce (Map.lookupIndex @k @a)
+
+findIndex :: forall k a. Ord k => k -> AppendMap k a -> Int
+findIndex = coerce (Map.findIndex @k @a)
+
+elemAt :: forall k a. Int -> AppendMap k a -> (k, a)
+elemAt = coerce (Map.elemAt @k @a)
+
+updateAt :: forall k a. (k -> a -> Maybe a) -> Int -> AppendMap k a -> AppendMap k a
+updateAt = coerce (Map.updateAt @k @a)
+
+deleteAt :: forall k a. Int -> AppendMap k a -> AppendMap k a
+deleteAt = coerce (Map.deleteAt @k @a)
+
+findMin :: forall k a. AppendMap k a -> (k, a)
+findMin = coerce (Map.findMin @k @a)
+
+findMax :: forall k a. AppendMap k a -> (k, a)
+findMax = coerce (Map.findMax @k @a)
+
+deleteMin :: forall k a. AppendMap k a -> AppendMap k a
+deleteMin = coerce (Map.deleteMin @k @ a)
+
+deleteMax :: forall k a. AppendMap k a -> AppendMap k a
+deleteMax = coerce (Map.deleteMax @k @a)
+
+deleteFindMin :: forall k a. AppendMap k a -> ((k, a), AppendMap k a)
+deleteFindMin = coerce (Map.deleteFindMin @k @a)
+
+deleteFindMax :: forall k a. AppendMap k a -> ((k, a), AppendMap k a)
+deleteFindMax = coerce (Map.deleteFindMax @k @a)
+
+updateMin :: (a -> Maybe a) -> AppendMap k a -> AppendMap k a
+updateMin f = coerce (Map.updateMin f)
+
+updateMax :: (a -> Maybe a) -> AppendMap k a -> AppendMap k a
+updateMax f = coerce (Map.updateMax f)
+
+updateMinWithKey :: (k -> a -> Maybe a) -> AppendMap k a -> AppendMap k a
+updateMinWithKey f = coerce (Map.updateMinWithKey f)
+
+updateMaxWithKey :: (k -> a -> Maybe a) -> AppendMap k a -> AppendMap k a
+updateMaxWithKey f = coerce (Map.updateMaxWithKey f)
+
+minView :: forall k a. AppendMap k a -> Maybe (a, AppendMap k a)
+minView = coerce (Map.minView @k @a)
+
+maxView :: forall k a. AppendMap k a -> Maybe (a, AppendMap k a)
+maxView = coerce (Map.maxView @k @a)
+
+minViewWithKey :: forall k a. AppendMap k a -> Maybe ((k, a), AppendMap k a)
+minViewWithKey = coerce (Map.minViewWithKey @k @a)
+
+maxViewWithKey :: forall k a. AppendMap k a -> Maybe ((k, a), AppendMap k a)
+maxViewWithKey = coerce (Map.maxViewWithKey @k @a)
+
+showTree :: forall k a. (Show k, Show a) => AppendMap k a -> String
+showTree = coerce (Map.showTree @k @a)
+
+showTreeWith :: (k -> a -> String) -> Bool -> Bool -> AppendMap k a -> String
+showTreeWith f = coerce (Map.showTreeWith f)
+
+valid :: forall k a. Ord k => AppendMap k a -> Bool
+valid = coerce (Map.valid @k @a)
+
+instance Default (AppendMap k a) where
+  def = empty
+
+instance AppendMap k' v' ~ t => Rewrapped (AppendMap k a) t
+
+instance Wrapped (AppendMap k a) where
+  type Unwrapped (AppendMap k a) = Map k a
+  _Wrapped' = iso (\(AppendMap m) -> m) AppendMap
+
+-- | Operator for creating a singleton 'Map'
+(=:) :: k -> a -> AppendMap k a
```

Comment:
Yes, but currently this is not exported by `Reflex.Dom`. Either way probably best to drop this to avoid any clashes in the future.